Bugsnag/Assets/Bugsnag/Editor/SymbolUpload/BugsnagCLI.cs

```diff
@@ -0,0 +1,184 @@
+        private readonly string DOWNLOADED_CLI_PATH = Path.Combine(Application.dataPath, "../bugsnag/bin/bugsnag_cli");
+        private readonly string DOWNLOADED_CLI_URL = $"https://github.com/bugsnag/bugsnag-cli/releases/download/v{DOWNLOADED_CLI_VERSION}/";
+        private readonly string _cliExecutablePath;
+        public BugsnagCLI()
+        {
+            var config = BugsnagSettingsObject.GetSettingsObject();
+
+            if (string.IsNullOrEmpty(config.BugsnagCLIExecutablePath))
+            {
+                _cliExecutablePath = DownloadBugsnagCLI();
+            }
+            else
+            {
+                _cliExecutablePath = config.BugsnagCLIExecutablePath;
+            }
+            if (!File.Exists(_cliExecutablePath))
+            {
+                throw new Exception($"BugSnag CLI not found at path: {_cliExecutablePath}");
+            }
+        }
+
+        public void UploadAndroidSymbols(string buildOutputPath, string apiKey, string versionName, int versionCode)
+        {
+            string args = $"upload unity-android --api-key={apiKey} --verbose --project-root={Application.dataPath} {buildOutputPath}";
+
+            if (!string.IsNullOrEmpty(versionName))
+                args += $" --version-name={versionName}";
+            if (versionCode > -1)
+                args += $" --version-code={versionCode}";
+
+            int exitCode = StartProcess(_cliExecutablePath, args, out string output, out string error);
+
+            if (exitCode != 0)
+            {
+                throw new Exception($"Error uploading symbols: {error}\nOutput: {output}");
+            }
+        }
+
+        public string DownloadBugsnagCLI()
+        {
+            EnsureDirectoryExists();
+            if (File.Exists(DOWNLOADED_CLI_PATH) && GetCurrentCliVersion() == DOWNLOADED_CLI_VERSION)
+            {
+                return DOWNLOADED_CLI_PATH;
+            }
+            DownloadCLI();
+            MakeExecutable();
+            return DOWNLOADED_CLI_PATH;
+        }
+
+        private void EnsureDirectoryExists()
+        {
+            var directory = Path.GetDirectoryName(DOWNLOADED_CLI_PATH);
+            if (!Directory.Exists(directory))
+            {
+                Directory.CreateDirectory(directory);
+            }
+        }
+
+        private void DownloadCLI()
+        {
+            string url = GetDownloadUrl();
+            if (string.IsNullOrEmpty(url))
+            {
+                throw new InvalidOperationException($"Unsupported platform: {RuntimeInformation.OSDescription}");
+            }
+
+            string tempPath = DOWNLOADED_CLI_PATH + ".tmp";
+
+            try
+            {
+                int exitCode = StartProcess("curl", $"-L {url} --output {tempPath}", out string output, out string error);
+                if (exitCode != 0)
+                {
+                    throw new InvalidOperationException($"Download failed: {error}");
```

Comment:
Suggest adding the path you were trying to download, unless it's going to be in the error from curl?

Reply:
yes it should be in the error if that path is the issue